### PR TITLE
IDEMPIERE-6117:POI: WorkbookFactory.create cannot be used to open XLS…

### PR DESCRIPTION
**Pull request by hieplq**
[IDEMPIERE-6117](https://idempiere.atlassian.net/browse/IDEMPIERE-6117)

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [x] I have tested the direct scenario that my code is solving
- [x] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [x] New and existing unit tests pass locally with my changes
- [x] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

[IDEMPIERE-6117]: https://idempiere.atlassian.net/browse/IDEMPIERE-6117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: org.idempiere.test/pom.xml [hunk 1]
```diff
@@ -133,6 +133,54 @@
...
+						<configuration>
+							<!--${url.file.srv}/database/11/Adempiere_pg.jar-->
+							<url>https://raw.githubusercontent.com/hieplq/binary.file/IDEMPIERE-6117/test/poi-ooxml-5.2.3.jar</url>
+							<unpack>false</unpack>
+							<outputDirectory>resources</outputDirectory>
+							<retries>5</retries>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
```
Comment (hengsin) [inline org.idempiere.test/pom.xml:183]: this is unnecessary, just add the test xlsx to test project. it is just 1 file and I don't expect we need to make frequent update of it either.

File: org.adempiere.base/src/org/adempiere/base/BaseActivator.java [hunk 2]
```diff
@@ -61,6 +143,9 @@ public void start(BundleContext context) throws Exception {
 		context.registerService(CommandProvider.class.getName(), new StackTraceCommand(), null);
 		
 		blacklistService = new ComponentBlackListService(context);
+		
+		int trackStates = Bundle.RESOLVED | Bundle.STOPPING | Bundle.ACTIVE | Bundle.STARTING;
+		new PoiBundleTracker(context, trackStates).open();
```
Comment (hengsin) [inline org.adempiere.base/src/org/adempiere/base/BaseActivator.java:148]: The example at https://github.com/apache/poi/blob/trunk/osgi/src/main/java/org/apache/poi/osgi/Activator.java just add the provider in activator, why do we need to use bundle tracker here ?

File: org.idempiere.test/pom.xml (hunk 1 above)
Comment (hieplq) [reply org.idempiere.test/pom.xml:183]: In addition to the XLSX file, I also have two JAR files for testing case updates and multi-version installation.

File: org.adempiere.base/src/org/adempiere/base/BaseActivator.java (hunk 2 above)
Comment (hieplq) [reply org.adempiere.base/src/org/adempiere/base/BaseActivator.java:148]: example for case merger POI and POI-OOXML into a single bundle

In Idempiere, I need the Bundle Tracker to ensure that both the wrapper bundle already resolved

File: org.idempiere.test/pom.xml (hunk 1 above)
Comment (hengsin) [reply org.idempiere.test/pom.xml:183]: but that's not what iDempiere core is using, we shouldn't add test here that's for your project work.

File: org.adempiere.base/src/org/adempiere/base/BaseActivator.java (hunk 2 above)
Comment (hengsin) [reply org.adempiere.base/src/org/adempiere/base/BaseActivator.java:148]: but that's not how it is configure in iDempiere core.

File: org.idempiere.test/pom.xml (hunk 1 above)
Comment (hengsin) [reply org.idempiere.test/pom.xml:183]: Also, for multi version, AFAIU, if you don't limit the import version in MANIFEST.MF, the resolver should always resolve to the latest version.

File: org.adempiere.base/src/org/adempiere/base/BaseActivator.java (hunk 2 above)
Comment (hengsin) [reply org.adempiere.base/src/org/adempiere/base/BaseActivator.java:148]: Unless there's compelling reason for doing so, I would prefer the simple approach in the link example above and don't have to depends on hard coded bundle name check as well.

Comment (hieplq) [reply org.adempiere.base/src/org/adempiere/base/BaseActivator.java:148]: I'd like a straightforward approach, similar to sample, but need to merged JARs for POI and POI-OOXML for deployment on the idempiere/binary.file

File: org.idempiere.test/pom.xml (hunk 1 above)
Comment (hengsin) [reply org.idempiere.test/pom.xml:183]: The test above doesn't test anything for multi version of POI. You install a different version and unless your bundle specifically import only that version, the OSGi resolver will just resolve to the latest version install.

File: org.adempiere.base/src/org/adempiere/base/BaseActivator.java (hunk 2 above)
Comment (hengsin) [reply org.adempiere.base/src/org/adempiere/base/BaseActivator.java:148]: If a plugin need a different version of POI, that should be managed in that plugin/bundle, it is not the core job to do that.

**Comment by CarlosRuiz-globalqss** [comment]
replaced by https://github.com/idempiere/idempiere/pull/2357

**Pull request closed without merging**